src/contribute/release-guide.md

```diff
@@ -189,30 +189,35 @@ Check out the version of the codebase from which you start the release. For a ne
 
 Set up a few environment variables to simplify Maven commands that follow. (We use `bash` Unix syntax in this guide.)
 
-    VERSION="1.2.3"
-    NEXT_VERSION="1.2.4"
-    BRANCH_NAME="release-${VERSION}"
-    DEVELOPMENT_VERSION="${NEXT_VERSION}-SNAPSHOT"
+    RELEASE_VERSION=2.5.0
+    NEXT_VERSION_IN_BASE_BRANCH=2.6.0
+    BRANCH=release-${RELEASE}
 
 Version represents the release currently underway, while next version specifies the anticipated next version to be released from that branch. Normally, 1.2.0 is followed by 1.3.0, while 1.2.3 is followed by 1.2.4.
 
-Use Maven release plugin to create the release branch and update the current branch to use the new development version. This command applies for the new major or minor version. (Warning: this command automatically pushes changes to the code repository.)
+**NOTE**: Only if you are doing an incremental/hotfix releasei (e.g. 2.5.1), please check out the previous release tag, before running the following instructions:
 
-    mvn release:branch \
-        -DbranchName=${BRANCH_NAME} \
-        -DdevelopmentVersion=${DEVELOPMENT_VERSION}
+    BASE_RELEASE=2.5.0
+    RELEASE=2.5.1
+    NEXT_VERSION_IN_BASE_BRANCH=2.5.0
+    git checkout tags/${BASE_RELEASE}
 
-However, if you are doing an incremental/hotfix release, please run the following command after checking out the release tag of the release being patched.
+Create a new branch, and update version files in the master branch.
 
-    mvn release:branch \
-        -DbranchName=${BRANCH_NAME} \
-        -DupdateWorkingCopyVersions=false \
-        -DupdateBranchVersions=true \
-        -DreleaseVersion="${VERSION}-SNAPSHOT"
+    git branch ${BRANCH}
```

Comment:
Do we need to clarify that the branch is to made on the upstream apache/beam repo and pushed, and not on a fork?

Reply:
It should be done in the upstream repo. I've mentioned 'Apache' there. LMK what you think.